### ISSUE TITLE
Offline puzzles not synced to rating

### ISSUE THREAD

**Issue by aronhegedus** [lichess-org/lichobile#2255]
**Describe the bug**
When doing the 50 offline puzzles, if I deliberately fail a lot in a row, then come back online, I don't lose any rating points, even when I wait for a while.

**To Reproduce**
1. Open lichess app
2. While online, navigate to `Puzzles`
3. Enable airplane mode to simulate going offline
4. Take a note of current rating
5. Wait until `50/50` text appears next to `Rating ? . Played X times`
6. Deliberately fail multiple puzzles, then click `view solution`, and then `continue training` to go to next
7. End airplane mode to come back online
8. Wait, then check your rating
9. The rating is the SAME as it was before failing all the puzzles.

Alternatively, you could solve a lot of offline puzzles, and it'd still not change the rating

**Expected behavior**
When coming back online after solving offline puzzles, there should be a sync with the server to update rating.

**Screenshots**

**Smartphone (please complete the following information):**
 - Device: iPhone SE
 - OS: iOS 16.1.1
 - Browser: safari, although this was on the app
 - Version: App is 7.16.0

**Additional context**
This should be taken care of somewhere in `src/ui/training/offlineService.ts`, not sure where specifically.

**Comment by larsen0815** [comment]
It will also not update the [history](https://lichess.org/training/history).

**Comment by finedust** [comment]
I point out that the bug is present also in the Android version. App version 8.0.0

**Comment by landry314** [comment]
Yes, the offline puzzles are not affecting the rating once you come back online. It just doesn't matter how well you have done, you have the same rating you had when you were online. This is a bug worth fixing. Doing chess puzzles while you have no signal is pretty awesome.
